### PR TITLE
Removed shadow from iOS screencaps

### PR DESCRIPTION
**Pull request by jankcat**
I have removed the shadow from iOS screencaps. 

The change essentially just takes slightly overlapping screenshots (scrolling to just above the previous screenshot's bottom edge, as opposed of directly at the edge of the previous screenshot), and then crops half the difference from the bottom of the first screenshot and from the top of the second screenshot.

I took the liberty of running it against my own suite as well, with the following results:

| Without Fix | With Fix | iPhone 6 iOS 9 | iPhone 7 iOS 10.2 |
|-|-|-|-|
|||| |

### CODE REVIEW

**Comment by zinserjan** [comment]
Awesome! Thanks for this.

Just two things before this is ready to merge:
- Can you add some tests for the calculation in TrimAndMergeScreenshotStrategy.js? For an example have a look at [MergeScreenshotStrategy.test.js](https://github.com/zinserjan/wdio-screenshot/blob/a567f5e8ef553d5f29f24b0e3d6482a5fe736ff9/test/unit/utils/strategies/MergeScreenshotStrategy.test.js)
- Can you also remove the comment about the grey line in the readme?

**Comment by jankcat** [comment]
For sure, let me take a look at that.

**Comment by jankcat** [comment]
Aaand saucelabs is down, so the int tests are failing: https://status.saucelabs.com/incidents/sczxwt2zwz30

**Comment by jankcat** [comment]
@zinserjan let me know if you want any more changes.

**Comment by jankcat** [comment]
@zinserjan Any more comments?

**Comment by zinserjan** [comment]
@jankcat sorry, hadn't any time to review this, yet.

Looks good for me. Great job! I think this is ready to merge. But before I'll give it a try on a few projects, probably on wednesday evening.

**Comment by jankcat** [comment]
Oh sure, fair enough. Let me know if it needs more work, I have time the end of this week to do so.

File: src/utils/strategies/TrimAndMergeScreenshotStrategy.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    // If first, do not trim off top.
+    if (y === 0) {
+      topTrim = 0;
+      // If first AND last, dont trim anything
+      if (!this.hasNextVerticalScrollPosition()) {
+        heightOffset = height;
+      } else {
+        heightOffset = height - NAV_SHADOW_CONST;
+      }
+    }
```
Comment (zinserjan) [inline src/utils/strategies/TrimAndMergeScreenshotStrategy.js]: Just thought if a `if/else-if/else-if/else` construct would make the cases more clearly. Maybe in the order "first AND last", "first only", "last only" and "between"?

Comment (jankcat) [inline src/utils/strategies/TrimAndMergeScreenshotStrategy.js]: Sure let me take a looksie

**Comment by jankcat** [comment]
@zinserjan I made the requested changes, I see what you mean about making it easier to understand.

**Comment by zinserjan** [comment]
@jankcat looks good. I'll merge this after the CI build is fixed (#74).

Question is how this should be released, as a patch (0.5.3) or as a minor/major (0.6.0)?  This PR is kind of breaking change in case of the capturing results :grin:

**Comment by jankcat** [comment]
Haha thats a tough one... From a regression standpoint I would say 0.6.0. No apis changed but the output would break visual diffing for sure.

**Comment by zinserjan** [comment]
I would also say 0.6.0 to avoid visual regressions.

**Comment by jankcat** [comment]
Hurray! Thanks.

**Comment by jankcat** [comment]
When will you do an npm publish?

**Comment by zinserjan** [comment]
wdio-screenshot 0.6.0 published :tada: Great job @jankcat :)

I'll publish also a new version of [wdio-visual-regression-service](https://github.com/zinserjan/wdio-visual-regression-service) :wink:

**Pull request merged**